book/src/dev/rfcs/0008-release-planning.md

```diff
@@ -0,0 +1,191 @@
+forever, or leave our users stranded.
+
+Instead, we can publish `zebrad` as a Rust crate and recommend `cargo install`.
+This approach has two major downsides.  First, installation takes longer,
+because Zebra is compiled locally.  Second, as long as we have a dependency on
+`zcashconsensus`, we'll have to instruct users to install some
+platform-specific equivalent of a `build-essential` package. And as long as
+we depend on `zcash_script`, we'll have to instruct users to install `libclang`.
+However, even for
+crates such as `zcashconsensus` that build native code, the `cargo`-managed
+build process is far, far more reliable than build processes for C or C++
+projects.  We would not be asking users to run autotools or `./configure`, just
+a one-step `cargo install`.  We also know that it's possible to reliably build
+Zebra on each platform with minimal additional steps, because we do so in our
+CI.
+
+In contrast to these downsides, distributing `zebra` through Cargo has a number
+of upsides.  First, because we distribute our libraries using crates.io, we
+already have to manage tooling for publishing to crates.io, so there's no
+additional work required to publish `zebrad` this way.  Second, we get a
+cross-platform update mechanism with no additional work, since `cargo install`
+will upgrade to the latest published version.  Third, we don't rely on any
+third parties to mediate the relationship between us and our users, so users
+can get updates as soon as we publish them.  Fourth, unlike a system package
+manager, we can pin exact hashes of every transitive dependency (via the
+`Cargo.lock`, which `cargo install` can be configured to respect).  Fifth,
+we're positioned to pick up (or contribute to) ecosystem-wide integrity
+improvements like a transparency log for `crates.io` or work on reproducible
+builds for Rust.
+
+This proposal is summarized above in the [guide-level
+explanation](#guide-level-explanation).
+
+## Release Processes
+
+The next question is what kind of release processes and automation we should
+use.  Here are two important priorities for these processes:
+
+1.  Reducing the friction of doing any individual release, allowing us to move
+    closer to a continuous deployment model;
+2.  Reducing the risk of error in the release process.
+
+These are roughly in order of priority but they're clearly related, since the
+more friction we have in the release process, the greater the risk of error,
+and the greater the risk of error, the more friction we require to prevent it.
+
+Automation helps to reduce friction and to reduce the risk of error, but
+although it helps enormously, it has declining returns after some point (in the
+sense that automating the final 5% of the work is significantly more complex
+and error-prone than automating the first 5% of the work).  So the challenge is
+to find the right "entry point" for the automated part of the system that
+strikes the right balance.
```

Comment:
Notes from the release discussion:

Should not be possible to do release from any one laptop (not technically impossible, not part of workflow and it is not ever done), requires levels of automation that provides resilience/ protects against bus factor, set up to make it difficult for one person to do it all/split among all engineers, should not be checklist, human runs a script

Reply:
On automating: reduces opportunity of mistakes, use a different process from writing new code and submitting PRs, so new code has to go through rest of process